src/mcts/search.cc

```diff
@@ -138,7 +138,10 @@ void Search::SendUciInfo() REQUIRES(nodes_mutex_) {
     const auto d = edge.GetD();
     const int w = static_cast<int>(std::round(500.0 * (1.0 + wl - d)));
     const auto q = edge.GetQ(default_q, draw_score);
-    if (score_type == "centipawn_with_drawscore") {
+    if (edge.IsTerminal() && wl != 0.0f) {
+      uci_info.mate = std::copysign(
+          edge.GetM(0.0f) / 2 + (edge.IsTbTerminal() ? 100 : 1), wl);
```

Comment:
Rounding mate score to the nearest integer before converting to integer should result in more accurate mate score for TB nodes for nets with moves left head support.

Add 101 instead of 100 for TB nodes to get +100 moves for TB nodes?

Reply:
Okay I'll switch it to `round(GetM / 2.0)`

As for 101 vs 100, I guess it's just someone's preference of whether a TB win should appear as M100 vs M101. Doesn't really matter much either way. Also I guess is 100 or 101 reasonable vs say M1000 or other?

Reply:
Was your intention to only report M100 for TB nodes? With moves left head TB nodes have estimated moves left. Adding 101 would report the mate score as 100 + estimated mate score. On TB positions the estimate should be quite accurate for certain TB nodes, I think it would be easier to parse that for TB nodes M123 means estimated 23 moves left.

Reply:
The behavior I intended and believe I implemented was plain adding 100 moves to TB positions, which with the network I've been testing with has no moves left head, so `GetM` for the converted ancestor is always ply depth to the true/TB terminal, which has a plies_left = 0. So the extra 100 moves is some generic "we need more moves because 0 plies left at the terminal is definitely inaccurate."

But I do see that with moves left head, the TB terminal inherits plies_left based on the parent's value… Which as you suggest would be reasonable to just directly use that `GetM` without the +100 approximation. So then how should we decide when to directly use that value vs approximate?

Reply:
I think the expectation of mate scores is that it's a proven mate. Using the moves left estimate could give too short estimate in some positions. I'm okay with adding 100 in those cases, but I wouldn't report mate score based on the estimate directly.

Certain and true terminal nodes have M that is exactly an integer but TB terminals don't. I was thinking of just rounding the `GetM`. It's the same as before for true terminals which I think worked correctly.

Reply:
Okay, rounding just M and adding 101 instead of 100 for TB.